extensions/amp-video/0.1/amp-video.js

```diff
@@ -47,13 +47,27 @@ class AmpVideo extends AMP.BaseElement {
      * @override
      */
     preconnectCallback(opt_onLayout) {
-      const videoSrc = this.element.getAttribute('src');
+      const videoSrc = this.getVideoSource_();
       if (videoSrc) {
         assertHttpsUrl(videoSrc, this.element);
         this.preconnect.url(videoSrc, opt_onLayout);
       }
     }
 
+    /**
+     * @private
+     */
+    getVideoSource_() {
+      let videoSrc = this.element.getAttribute('src');
+      if (!videoSrc) {
+        const srcElements = this.element.getElementsByTagName('source');
+        if (srcElements) {
+          videoSrc = srcElements[0].getAttribute('src');
```

Comment:
you can use `elementByTag` from `dom.js` which handles returning first or `null`

```
const source = elementByTag(this.element, `source`);
if (source) {
  return source.getAttribute('src');
}
```